activation/handler_v2.go

```diff
@@ -682,6 +682,19 @@ func (h *HandlerV2) checkMalicious(ctx context.Context, tx sql.Transaction, atx
 	if malicious {
 		return true, nil
 	}
+	if atx.MarriageATX != nil {
+		receivedProof, err := identities.IsMarriageMalicious(tx, *atx.MarriageATX)
+		if err != nil {
+			return false, fmt.Errorf("checking if marriage ATX is malicious: %w", err)
+		}
+		if receivedProof != nil {
+			err = identities.SetMaliciousBecauseOfMarriage(tx, atx.SmesherID, *receivedProof)
+			if err != nil {
+				return false, fmt.Errorf("setting node as malicious because of marriage: %w", err)
+			}
+			return true, nil
+		}
+	}
```

Comment:
I think this might not be sufficient. The marriage ATX (the one that marries IDs - it includes the marriage certificates) must also be marked as malicious when it is processed if any of the married IDs is already malicious. This was previously covered by matching by marriage ATX in the `IsMalicious()` func.

Reply:
so there are 2 things now:
- all identities with same marriage_atx that are present in db at the moment of SetMalicious are marked malicious
- if some identity is learned after SetMalicious and it has same marriage we update it here as malicious

it doesn't cover the case you described?

Reply:
so the case is MarriageAtx is nil, so that it doesn't point to itself?
but instead it has some certificates, does it work if pick id of such atx as marriage_atx and run everything the same way?

although i don't see that covered in this query

```
	SELECT 1 FROM identities
	WHERE (
		marriage_atx = (SELECT marriage_atx FROM identities WHERE pubkey = ?1 AND marriage_atx IS NOT NULL)
		AND proof IS NOT NULL
	)
	OR (pubkey = ?1 AND marriage_atx IS NULL AND proof IS NOT NULL);
```

Reply:
i feel stupid there is even a failing test and i can't figure what is going on

Reply:
The problem is that an ATX that contains a marriage and is syntactically valid marks all identities in the marriage set as malicious if at least one identity in the set is already malicious.

I do have a fix for the problem but it requires calling `IsMalicious` on all identities in a marriage ATX before updating the marriage set (which I think is OK, it should be much faster then inserting the knowledge about the set anyway)